### DISCUSSION TITLE
Supabase error:  delayed connect error: 111 / operator does not exist: uuid = text

### DISCUSSION THREAD

**Discussion by charelF** [supabase/supabase#26769] [Questions] [answered]
When creating a new user in my remote project, I get the following error:

`upstream connect error or disconnect/reset before headers. reset reason: remote connection failure, transport failure reason: delayed connect error: 111`

I do not get this error in my local project. Because of that, in my local project (which is linked), I did `npx supabase db reset --linked` and also `npx supabase db push` hoping it would fix my remote project but it did not.

I get the above error when trying to manually insert a new user into the auth table via the dashboard:

<img width="464" alt="image" src="https://github.com/supabase/supabase/assets/29770094/a56a4513-70b7-4a7f-933c-77d1c36263db">

In the postgress logs, I have also noticed this: `operator does not exist: uuid = text`

<img width="611" alt="image" src="https://github.com/supabase/supabase/assets/29770094/c6613f53-4a15-4211-aca8-925e6fdf1019">

Full message: 

```
Log ID
26203db1-71e6-498f-82ee-c35574e688c0

Log Timestamp (UTC)
2024-05-26T09:08:56.444Z

Log Event Message
operator does not exist: uuid = text

Log Metadata
[
  {
    "file": null,
    "host": "db-fvjjxygvmzebsvmrqfyk",
    "metadata": [],
    "parsed": [
      {
        "application_name": "gotrue_migrations",
        "backend_type": "client backend",
        "command_tag": "DO",
        "connection_from": "127.0.0.1:48080",
        "context": "PL/pgSQL function inline_code_block line 3 at SQL statement",
        "database_name": "postgres",
        "detail": null,
        "error_severity": "ERROR",
        "hint": "No operator matches the given name and argument types. You might need to add explicit type casts.",
        "internal_query": "update auth.identities\n  set last_sign_in_at = '2022-11-25'\n  where\n    last_sign_in_at is null and\n    created_at = '2022-11-25' and\n    updated_at = '2022-11-25' and\n    provider = 'email' and\n    id = user_id::text",
        "internal_query_pos": 203,
        "leader_pid": null,
        "location": null,
        "process_id": 124012,
        "query": "-- previous backfill migration left last_sign_in_at to be null, which broke some projects\n\ndo $$\nbegin\nupdate auth.identities\n  set last_sign_in_at = '2022-11-25'\n  where\n    last_sign_in_at is null and\n    created_at = '2022-11-25' and\n    updated_at = '2022-11-25' and\n    provider = 'email' and\n    id = user_id::text;\nend $$;\n",
        "query_id": 9143148468488802000,
        "query_pos": null,
        "session_id": "6652fc28.1e46c",
        "session_line_num": 3,
        "session_start_time": "2024-05-26 09:08:56 UTC",
        "sql_state_code": "42883",
        "timestamp": "2024-05-26 09:08:56.444 UTC",
        "transaction_id": 0,
        "user_name": "supabase_auth_admin",
        "virtual_transaction_id": "9/3777"
      }
    ],
    "parsed_from": null,
    "project": null,
    "source_type": null
  }
]
```

In the overall logs I also see errors in my health endpoint:

<img width="612" alt="image" src="https://github.com/supabase/supabase/assets/29770094/45ff5b8e-316e-45cd-abc0-8674a4df9fce">

This seems like something about my database setup is wrong, but I cant figure out what, especially since locally everything seems to work. Here are my only 2 migration files, which should also be the ones behind my remote database?

```sql
create type public.app_role as enum ('admin', 'agent');

create table public.profiles (
  id uuid references auth.users not null primary key,
  updated_at timestamp with time zone,
  full_name text,
  role app_role not null default 'agent'
);
-- Set up Row Level Security (RLS)
alter table profiles enable row level security;

create policy "Allow authenticated all access" on public.profiles for all using ( auth.role() = 'authenticated' );

create function public.handle_new_user()
returns trigger as $$
begin
  insert into public.profiles (id)
  values (new.id);
  return new;
end;
$$ language plpgsql security definer;

create trigger on_auth_user_created
  after insert on auth.users
  for each row execute procedure public.handle_new_user();
```

```sql
-- jobs can be in 3 states: 
create type public.job_state as enum ('requested', 'ready', 'active', 'finished');

-- Create a table for jobs
create table public.jobs (
  id uuid primary key,
  -- Form metadata
  submission_date timestamp with time zone,
  form_title text,
  updates int,

  -- Reference to the user assigned to this job
  assigned_user_id uuid references profiles(id),

  -- job state
  state job_state not null,
  
  -- Client information
  client_name text,
  client_address_1 text,
  client_address_2 text,
  client_tel text,
  client_email text,

  -- Chantier information
  chantier_name text,
  chantier_address_1 text,
  chantier_address_2 text,
  chantier_tel text,
  chantier_email text,

  -- RDV information
  meeting_date date,
  meeting_hour time,
  call_before time,

  -- Problem description
  fenetre int,
  coulissant int,
  vitrage_fissure int,
  porte_fenetre int,
  porte_entree int,
  autre text,
  client_message text,
  photo_link_1 text,
  photo_link_2 text,
  photo_link_3 text,

  -- raw original request
  request jsonb
);

-- Set up Row Level Security (RLS)
alter table jobs enable row level security;

create policy "Allow authenticated all access" on public.jobs for all using ( auth.role() = 'authenticated' );
```

Any leads on what the issue could be / or how i could debug this further remotely given that I cant reproduce it locally?

**Comment by charelF** [comment]
I do believe there is a link between this message in my above log: `update auth.identities\n  set last_sign_in_at = '2022-11-25'\n  where\n    last_sign_in_at is null and\n    created_at = '2022-11-25' and\n    updated_at = '2022-11-25' and\n    provider = 'email' and\n    id = user_id::text"` particularly **id = user_id::text"** and  this link: https://github.com/orgs/supabase/discussions/7697#discussioncomment-3151524

but i cant figure out how who sends this query and how i could fix it

**Reply by charelF** [reply]
digging more into it, this is the migration that causes it - https://github.com/supabase/auth/blob/b954a485096cddfd1eef4d582034a99eff95fa6f/migrations/20221208132122_backfill_email_last_sign_in_at.up.sql#L4

**Comment by charelF** [comment]
So nothing i did solved it
- resetting the remote database
- completely deleting all migrations (empty migrations folder) then resetting database such that nothing was set up
- restarting (complete restart) of the project

nothing solved it, the error remained

What i ended up doing that "solved my issue" deleting the (remote) project and making a new one, linking my local setup to this new remote project and the error was magically solved

I would still very much appreciate if anyone has an idea how i ended up in a state where my auth database was corrupted? Im worried this could happen again in a moment where i can not just easily delete the project

**Comment by GaryAustin1** [comment, marked as answer]
Probably related to this: https://github.com/orgs/supabase/discussions/20722

**Reply by charelF** [reply]
yes i did not see that discussion before i posted but they do seem very much related. thanks for linking

**Reply by tananga** [reply]
It also solved my problem :)
